pinecone/models/collections.go

```diff
@@ -56,27 +52,18 @@ func (model *CollectionResourceModel) Read(collection *pinecone.Collection) {
 	model.Source = types.StringValue(model.Source.ValueString())
 	model.Status = types.StringValue(string(collection.Status))
 	model.Environment = types.StringValue(collection.Environment)
-	if collection.Size != nil {
-		model.Size = types.Int64Value(*collection.Size)
-	}
-	if collection.Dimension != nil {
-		model.Dimension = types.Int64Value(int64(*collection.Dimension))
-	}
-	// var vectorCount *int64
-	// if collection.VectorCount != nil {
-	// 	tempVectorCount := int64(*collection.VectorCount)
-	// 	vectorCount = &tempVectorCount
-	// }
-	// model.VectorCount = types.Int64PointerValue(vectorCount)
+	model.Size = types.Int64Value(collection.Size)
+	model.Dimension = types.Int32Value(collection.Dimension)
+	model.VectorCount = types.Int32Value(collection.VectorCount)
```

Comment:
These are technically optional from the backend perspective, but they're typed as values and not pointers in the Go SDK. Sort of made me rethink that typing, but I think just letting them default to the zero-value is probably fine. Let me know if this seems egregious though.

Reply:
These are result values from the API, correct? The terraform provider is never submitting a zero-value to the server?

Reply:
Yeah, this is converting the incoming `Collection` model from the SDK when it's `Read`. I think the `Create` methods are what are actually handling reading in the terraform plan data, and sending the request: https://github.com/pinecone-io/terraform-provider-pinecone/blob/main/pinecone/provider/index_resource.go#L220-L260